### PR TITLE
Add new `Minitest/TestCaseName` cop

### PR DESCRIPTION
**Pull request by fatkodima**
This cop enforces that public test case method names start with `test_` prefix.

Missed `test_` prefix is almost always an unintentional mistake - as you can see from 2 corrected tests in this PR - https://github.com/rubocop-hq/rubocop-minitest/pull/89/files#diff-50588427db1de725ce02fc0d20cee56aR88 and https://github.com/rubocop-hq/rubocop-minitest/pull/89/files#diff-f64b784295c232ad3b46ae3d9a8808acR88.

### CODE REVIEW

File: lib/rubocop/cop/minitest/test_case_name.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      #   end
+      #
+      #   # good
+      #   class FooTest < Minitest::Test
+      #     def test_does_something
+      #       assert_equal 42, do_something
+      #     end
+      #   end
+      #
+      class TestCaseName < Cop
```
Comment (koic) [inline lib/rubocop/cop/minitest/test_case_name.rb:23]: I think that `TestMethodName` may be clearer.

File: lib/rubocop/cop/minitest/test_case_name.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        include MinitestExplorationHelpers
+        include DefNode
+
+        MSG = 'Test case name should start with `test_` prefix.'
```
Comment (koic) [inline lib/rubocop/cop/minitest/test_case_name.rb:27]:
```suggestion
        MSG = 'Test method name should start with `test_` prefix.'
```

File: lib/rubocop/cop/mixin/minitest_exploration_helpers.rb [hunk 3]
```diff
@@ -47,6 +47,15 @@ module MinitestExplorationHelpers
         refute_same
       ].to_set.freeze
 
+      HOOKS = %i[
```
Comment (koic) [inline lib/rubocop/cop/mixin/minitest_exploration_helpers.rb:50]:
```suggestion
      LIFECYCLE_HOOK_METHODS = %i[
```

File: lib/rubocop/cop/mixin/minitest_exploration_helpers.rb [hunk 4]
```diff
@@ -74,6 +83,10 @@ def test_cases(class_node)
       def assertion?(node)
         node.send_type? && ASSERTIONS.include?(node.method_name)
       end
+
+      def hook?(node)
```
Comment (koic) [inline lib/rubocop/cop/mixin/minitest_exploration_helpers.rb:87]:
```suggestion
      def lifecycle_hook_method?(node)
```

**Comment by fatkodima** [comment]
Updated.

**Comment by koic** [comment]
Thanks!

**Pull request merged**